geoh5py/shared/utils.py

```diff
@@ -180,3 +181,118 @@ def iterable_message(valid: list[Any] | None) -> str:
         msg = f" Must be: '{valid[0]}'."
 
     return msg
+
+
+KEY_MAP = {
+    "values": "Data",
+    "cells": "Cells",
+    "surveys": "Surveys",
+    "trace": "Trace",
+    "trace_depth": "TraceDepth",
+    "vertices": "Vertices",
+    "octree_cells": "Octree Cells",
+    "property_groups": "PropertyGroups",
+    "u_cell_delimiters": "U cell delimiters",
+    "v_cell_delimiters": "V cell delimiters",
+    "z_cell_delimiters": "Z cell delimiters",
+    "color_map": "Color map",
+    "metadata": "Metadata",
+    "options": "options",
+    "concatenated_object_ids": "Concatenated object IDs",
+    "concatenated_attributes": "Attributes",
+    "property_group_ids": "Property Group IDs",
+}
+
+
+def is_uuid(value: str):
+    """Check if a string is UUID compliant."""
+    try:
+        UUID(str(value))
+        return True
+    except ValueError:
+        return False
+
+
+def entity2uuid(value):
+    """Convert an entity to its UUID."""
+    if hasattr(value, "uid"):
+        return value.uid
+    return value
+
+
+def uuid2entity(value, workspace):
```

Comment:
being non-specific on the input value allows us to move the logic here instead of repeating it in the code. Basically returns the value as is if it can't transform.